lib/vector-core/buffers/src/buffer_usage_data.rs

```diff
@@ -0,0 +1,87 @@
+use core_common::internal_event::emit;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::{sync::Arc, time::Duration};
+use tokio::time::interval;
+use tracing::{Instrument, Span};
+
+use crate::internal_events::{BufferEventsReceived, BufferEventsSent, EventsDropped};
+
+pub struct BufferUsageData {
+    received_event_count: AtomicUsize,
+    received_event_byte_size: AtomicUsize,
+    sent_event_count: AtomicUsize,
+    sent_event_byte_size: AtomicUsize,
+    dropped_event_count: Option<AtomicUsize>,
```

Comment:
What is the difference between a `dropped_event_count` that is `None` and `Some(0)`? If we don't have a distinction between the two you might consider dropping the `Option`.

Reply:
I intended `None` to represent a non-dropping buffer whereas `Some(0)` would be a dropping buffer that hasn't yet dropped an event. We also avoid emitting the `buffer_discarded_events_total` metric under the `None` case. Let me know if that makes sense.